### PR TITLE
Allow overriding of any graph properties

### LINKED ISSUE #21: Add in ability to disable annotations
There doesn't seem to be a way to disable annotations.  Setting `enable: false` on an annotation dictionary has no effect and Dashboard seems to have an unused global `enable` option - https://github.com/uber/grafana-dash-gen/blob/master/grafana/dashboard.js#L85.  Perhaps something like #3 applied to https://github.com/uber/grafana-dash-gen/blob/master/grafana/annotations/graphite.js#L50 would be the best fix.

### PR DESCRIPTION
**Pull request by Willyham**
@MadanThangavelu 

Fixes #21

### CODE REVIEW

**Comment by eculver** [comment]
I recently bumped up against this trying to override axis units. I would be open to making this possible, but I think we should document all of state and perhaps even provide explicit interfaces for setting nested properties of the dashboard and panels.

**Comment by eculver** [comment]
I have since become a believer in this and I think it should invalidate your other PR, #9, should we merge it. However, I still think it's important to document what's going on when passing in a big configuration object. Specifically, the internal `state` object should be laid out in detail and what its defaults are and what can be overridden.

I would be willing to dive in and write the documentation, but I think that's all that we need before we merge this as part of 1.x and 2.x.

**Comment by Willyham** [comment]
I agree Evan. If we choose to document it with JS doc it may be possible to generate a lot of the grunt work. However, it may also be best just to document any additions and refer to Grafana documentation for all of the graph related options. That way, we wont be outdated when upgrades to grafana change/add options.

That being said, I can't find any official documentation of the dashboard json format :(

**Comment by albertyw** [comment]
I've rebased this PR to fix merge conflicts and because I'd also like to use the functionality in here.  I'll also be adding the same ability to annotations.

**Comment by albertyw** [comment]
@eculver @MadanThangavelu - Looks like Seung and I have both duplicated work that was already done in this PR.  Bump

**Comment by albertyw** [comment]
Rebased again to fix merge conflicts.

**Comment by albertyw** [comment]
@jwyuan

File: grafana/annotations/graphite.js [hunk 1]
```diff
@@ -20,6 +20,8 @@
 
 'use strict';
 
+var _ = require('underscore');
```
Comment (MadanThangavelu) [inline grafana/annotations/graphite.js]: Can we remove _ as a dependency?

File: grafana/annotations/graphite.js [hunk 2]
```diff
@@ -20,6 +20,8 @@
...
+var defaultsDeep = require('defaults-deep');
```
Comment (MadanThangavelu) [inline grafana/annotations/graphite.js]: Can we not just get away by using xtend here aswell?

**Comment by albertyw** [comment]
@MadanThangavelu Instead of using `xtend`, I just copied the logic from #24 and went for vanilla JS without dependencies.

**Comment by eculver** [comment]
@Willyham that makes two of us who have looked for documentation and failed. I resorted to reading the [Grafana source](https://github.com/grafana/grafana/blob/master/pkg/api/dashboard.go) to reason about it.

Regardless, all the tests pass, but we were missing a few bits of coverage, so I added them. Once CI finishes, I'll merge and cut a new minor.

Thanks for your help on this guys.

**Comment by eculver** [comment]
Will release as v2.2.0 -- and more docs coming.

**Comment by albertyw** [comment]
Thanks @Willyham @eculver @MadanThangavelu !

**Pull request merged**